### PR TITLE
Deploy testing cluster for both master and v0-4

### LINKED ISSUE #310: Make auto-deploy deploy based on version number plus HEAD
Related #95

### PR DESCRIPTION
**Pull request by gabrielwen**
Fixes #310 

Updates codes to be able to take as input branch names and use it for snapshot and deployment.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/313)

### CODE REVIEW

File: test-infra/auto-deploy/deploy-cron-head.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: auto-deploy-head
```
Comment (richardsliu) [inline test-infra/auto-deploy/deploy-cron-head.yaml]: auto-deploy-master?

Comment (gabrielwen) [reply test-infra/auto-deploy/deploy-cron-head.yaml]: Updated.

**Pull request merged**